app/src/ipc.js

```diff
@@ -1,24 +1,140 @@
 const { contextBridge, ipcRenderer } = require('electron');
 
+/*
+  Hardware wallet IPC channels
+  These const need to be the same as in ../../libs/hardwareWallet/ledger/constants
+  For security reasons (sandboxing) we cannot import them in this preload script
+*/
+const REQUEST = 'REQUEST';
+const RESPONSE = 'RESPONSE';
+const GET_SIGNED_MESSAGE = 'GET_SIGNED_MESSAGE';
+const GET_SIGNED_TRANSACTION = 'GET_SIGNED_TRANSACTION';
+const GET_PUB_KEY = 'GET_PUB_KEY';
+const RESET_LEDGER_IPC_QUEUE = 'RESET_LEDGER_IPC_QUEUE';
+const GET_CONNECTED_DEVICES = 'GET_CONNECTED_DEVICES';
+const LEDGER_HW_HID_EVENT = 'LEDGER_HW_HID_EVENT';
+/*
+  END Hardware wallet IPC channels
+*/
+
+/*
+  Global IPC channels
+  These const need to be the same as in .src/const/ipcGlobal
+  For security reasons (sandboxing) we cannot import them in this preload script
+*/
+const IPC_UPDATE_AVAILABLE = 'IPC_UPDATE_AVAILABLE';
+const IPC_UPDATE_STARTED = 'IPC_UPDATE_STARTED';
+const IPC_DOWNLOAD_UPDATE_START = 'IPC_DOWNLOAD_UPDATE_START';
+const IPC_DOWNLOAD_UPDATE_PROGRESS = 'IPC_DOWNLOAD_UPDATE_PROGRESS';
+const IPC_DOWNLOAD_UPDATE_COMPLETED = 'IPC_DOWNLOAD_UPDATE_COMPLETED';
+const IPC_UPDATE_QUIT_AND_INSTALL = 'IPC_UPDATE_QUIT_AND_INSTALL';
+const IPC_OPEN_URL = 'IPC_OPEN_URL';
+const IPC_STORE_CONFIG = 'IPC_STORE_CONFIG';
+const IPC_CONFIG_RETRIEVED = 'IPC_CONFIG_RETRIEVED';
+const IPC_RETRIEVE_CONFIG = 'IPC_RETRIEVE_CONFIG';
+const IPC_SET_LOCALE = 'IPC_SET_LOCALE';
+const IPC_DETECT_LOCALE = 'IPC_DETECT_LOCALE';
```

Comment:
reuse from src/const/ipcGlobal?

Reply:
Unfortunately we must have duplicate consts for the IPC channel const since electron doesn't allow imports into preload scripts (which our ipc is) in newer versions of electron https://github.com/electron/electron/issues/35587